build.zig

```diff
@@ -22,76 +22,42 @@ pub fn build(b: *std.Build) void {
         .root = upstream.path("."),
         .files = &[_][]const u8{
             "db/builder.cc",
-            "db/builder.h",
```

Comment:
Compiling header files as C source is not recommended approach. 

It also cause build failure for windows https://github.com/ChainSafe/leveldb.zig/actions/runs/18096857267/job/51489821068